nav2_util/include/nav2_util/geometry_utils.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_UTIL__GEOMETRY_UTILS_HPP_
+#define NAV2_UTIL__GEOMETRY_UTILS_HPP_
+
+#include "geometry_msgs/msg/quaternion.hpp"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.h"
+
+namespace nav2_util
+{
+namespace geometry_utils
+{
+
+inline geometry_msgs::msg::Quaternion orientationAroundZAxis(double angle)
```

Comment:
Adding an entire file for this function seems overkill for something that's used in only one place. 
I would either:
1) just create the local function where it's needed (AMCL)
2) see if this can be upstreamed into tf2_geometry_msgs as a more general "angleToQuat" function with x, y and z inputs

Reply:
@mkhansen-intel there used to be a function like that in TF1 and it was intentionally removed from TF2 for reasons that I personally don't agree with.

I agree having these utilities that aren't really used anywhere is getting a little out of hand. Much of the `nav2_utils` package consists of things only used in 1 package, that while could be used by other people, are generally simple enough to not really warrant it (stripping slashes, wrappers on angles package, etc). The lifecycle / node / service / action wrappers make sense to pull out of a specific package, but that seem to be to be more of a `nav2_common` type contribution over `utils`

Reply:
@mkhansen-intel @SteveMacenski I updated the nav2 stack to fully use this new function. I'll leave it up to you two if you want to keep it this way or remove the function to a local function. 

I should point out I found about 3 other exact implementations of this as a local function in various places.